### PR TITLE
Addon code alignments

### PR DESCRIPTION
**Pull request by FinalAngel**
* Added test matrix
* Adapted code base to align with other supported addons
## General

- [x] Update changelog
Adapted code base to align with other supported addons
- [x] Add contributors text to README file
We're grateful to all contributors who have helped create and maintain this package. Contributors are listed at [link to https://github.com/divio/djangocms-video/graphs/contributors].
- [x] Check that python, django, django-cms and other versions are listed correctly (add badge)
- [x] Make sure Django is not pinned in setup.py
- [x] Align gitignore files
- [x] Align setup.py
- [x] Keep the two line rule

## Tests

- [x] Make sure naming for the test files is test_.py
- [x] Add tests_migration.py file if missing
- [x] Remove test migration check in other files
- [x] Check requirements.txt file for accuracy (e.g. unpin djangocms-helper)
- [x] Check settings.py that it is ordered correctly
- [x] Added isort and updated files

## Internal

- [x] Run internal checklist

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,64 +1,93 @@
...
-    ) + extra_dependencies,
-    entry_points='''
-        [console_scripts]
-        aldryn-django=aldryn_django.cli:main
-    ''',
+    url='https://github.com/divio/aldryn-django',
+    license='BSD',
+    description='An opinionated Django setup bundled as an Aldryn Addon',
+    long_description=open('README.rst').read(),
+    packages=find_packages(exclude=['tests']),
```
Comment (bplociennik) [inline setup.py:88]: Are you adding this for further tests(except test_migrations)? Do we need this for aldryn-django?

File: tests/test_migrations.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
```
Comment (vThaian) [inline tests/test_migrations.py]: Do we need this test here?

File: tox.ini [hunk 3]
```diff
@@ -0,0 +1,62 @@
+[tox]
+envlist =
+    flake8
+    isort
+    py{27,34,35,36}-dj111-cms{34,35,36}
```
Comment (bplociennik) [inline tox.ini]: we don't need this for aldryn-django 2.0

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+matrix:
+  include:
+    - python: 3.5
+      env: TOX_ENV='flake8'
+    - python: 3.5
+      env: TOX_ENV='isort'
+    # Django 2.0
+    - python: 3.5
+      env: DJANGO='dj20' CMS='cms36'
```
Comment (bplociennik) [inline .travis.yml:13]: Please add more jobs here for other python versions supported in Django 2.0
https://docs.djangoproject.com/en/2.1/releases/2.0/#python-compatibility

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+install:
+  - pip install coverage isort tox
+  - "if [[ $TRAVIS_PYTHON_VERSION == '2.7' ]]; then export PY_VER=py27; fi"
```
Comment (bplociennik) [inline .travis.yml:17]: we don't need 2.7 python for Django 2.0

File: setup.py [hunk 6]
```diff
@@ -1,64 +1,94 @@
...
+CLASSIFIERS = [
+    'Development Status :: 5 - Production/Stable',
+    'Environment :: Web Environment',
+    'Intended Audience :: Developers',
+    'License :: OSI Approved :: BSD License',
+    'Operating System :: OS Independent',
+    'Framework :: Django',
+    'Framework :: Django :: 1.11',
+    'Framework :: Django :: 2.0',
+    'Framework :: Django :: 2.1',
```
Comment (bplociennik) [inline setup.py:64]: to remove

File: setup.py [hunk 7]
```diff
@@ -1,64 +1,94 @@
...
+
+
+CLASSIFIERS = [
+    'Development Status :: 5 - Production/Stable',
+    'Environment :: Web Environment',
+    'Intended Audience :: Developers',
+    'License :: OSI Approved :: BSD License',
+    'Operating System :: OS Independent',
+    'Framework :: Django',
+    'Framework :: Django :: 1.11',
```
Comment (bplociennik) [inline setup.py:62]: to remove

File: setup.py [hunk 8]
```diff
@@ -1,64 +1,94 @@
...
+    'Framework :: Django :: 2.0',
+    'Framework :: Django :: 2.1',
+    'Programming Language :: Python',
+    'Programming Language :: Python :: 2',
```
Comment (bplociennik) [inline setup.py:66]: to remove

File: setup.py [hunk 9]
```diff
@@ -1,64 +1,94 @@
...
+    'Programming Language :: Python :: 2.7',
```
Comment (bplociennik) [inline setup.py:67]: to remove

File: setup.py [hunk 10]
```diff
@@ -1,64 +1,94 @@
...
+    'Programming Language :: Python :: 3',
+    'Programming Language :: Python :: 3.4',
+    'Programming Language :: Python :: 3.5',
+    'Programming Language :: Python :: 3.6',
```
Comment (bplociennik) [inline setup.py:71]: add also python 3.7

File: tests/test_migrations.py (hunk 2 above)
Comment (bplociennik) [inline tests/test_migrations.py:1]: do we need this tests for aldryn-django? We don't use here any migrations

File: tox.ini [hunk 11]
```diff
@@ -0,0 +1,58 @@
+[tox]
+envlist =
+    flake8
+    isort
+    py{34,35,36}-dj20-cms36
```
Comment (bplociennik) [inline tox.ini:5]: please add also python 3.7

File: tox.ini [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+sections = FUTURE, STDLIB, DJANGO, CMS, THIRDPARTY, FIRSTPARTY, LIB, LOCALFOLDER
+known_first_party = aldryn_django
+known_cms = cms, menus
+known_django = django
+
+[testenv]
+deps =
+    -r{toxinidir}/tests/requirements.txt
+    dj20: Django>=2.0,<2.1
+    cms34: django-cms>=3.4,<3.5
```
Comment (bplociennik) [inline tox.ini:42]: it doesn't support django 2.0

File: tox.ini [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+    cms35: django-cms>=3.5,<3.6
```
Comment (bplociennik) [inline tox.ini:43]: it doesn't support django 2.0

**Pull request merged**